src/TTextEdit.cpp

```diff
@@ -1160,28 +1162,43 @@ void TTextEdit::mousePressEvent(QMouseEvent* event)
 
 
         QAction* action = new QAction("copy", this);
-        action->setStatusTip(tr("copy selected text to clipboard"));
+        action->setToolTip(tr("copy selected text to clipboard"));
         connect(action, SIGNAL(triggered()), this, SLOT(slot_copySelectionToClipboard()));
         QAction* action2 = new QAction("copy HTML", this);
-        action2->setStatusTip(tr("copy selected text with colors as HTML (for web browsers)"));
+        action2->setToolTip(tr("copy selected text with colors as HTML (for web browsers)"));
         connect(action2, SIGNAL(triggered()), this, SLOT(slot_copySelectionToClipboardHTML()));
         QAction* action3 = new QAction("select all", this);
-        action3->setStatusTip(tr("select all text in the buffer"));
+        action3->setToolTip(tr("select all text in the buffer"));
         connect(action3, SIGNAL(triggered()), this, SLOT(slot_selectAll()));
 
         QString selectedEngine = mpHost->getSearchEngine().first;
         QAction* action4 = new QAction(("search on " + selectedEngine), this);
         connect(action4, SIGNAL(triggered()), this, SLOT(slot_searchSelectionOnline()));
-        action4->setStatusTip("search selected text on " + selectedEngine);
+        action4->setToolTip(tr("search selected text on %1").arg(selectedEngine));
 
         auto popup = new QMenu(this);
+        popup->setToolTipsVisible(true); // Not the default...
         popup->addAction(action);
         popup->addAction(action2);
         popup->addSeparator();
         popup->addAction(action3);
         popup->addSeparator();
         popup->addAction(action4);
 
+        if (mudlet::self()->isControlless()) {
+            QAction* actionRestoreMainMenu = new QAction(tr("restore Main menu"), this);
+            connect(actionRestoreMainMenu, SIGNAL(triggered()), mudlet::self(), SLOT(slot_restoreMainMenu()));
+            actionRestoreMainMenu->setToolTip(tr("use this to restore the Main menu to get access to controls!"));
```

Comment:
Nothing to get excited about here, I'd recommend removing the exclamation mark.

Reply:
I'd be a little excited if I've accidentally set the options so that I cannot get to the controls (including the control to get the access back) and then found this _recovery_ option that means I do not have to abort my current session just to restore them... but okay.

BTW How long does a tool-tip have to be to be a sentence?